parlai/core/torch_agent.py

```diff
@@ -433,6 +439,13 @@ def add_cmdline_args(cls, argparser):
         agent.add_argument(
             '--fp16', type='bool', default=False, help='Use fp16 computations.'
         )
+        agent.add_argument(
+            '--fp16-impl',
+            type=str,
+            default='apex',
+            choices=['apex', 'mem_efficient'],
```

Comment:
so we technically don't even need apex anymore to run fp16?

Reply:
technically. but you probably only want to use mem_efficient in the case where memory is a concern. the memory efficient version can be a bit unstable, and currently only supports a couple of optimizers (adam, adafactor)